blas/testblas/level1double.go

```diff
@@ -1971,6 +2100,22 @@ func DrotmgTest(t *testing.T, d Drotmger) {
 		if !dTolEqual(rx1, test.Rx1) {
 			t.Errorf("drotmg rx1 mismatch %v: expected %v, found %v", test.Name, test.Rx1, rx1)
 		}
+
+		// Drotmg routines compute the components of a modified Givens transformation
+		// matrix H that zeros the y-component of the resulting vector,
+		//
+		//  [x1; 0] := H[x1 sqrt(d1); y1 sqrt(d2)].
+		//
+		// Drotm performs a modified Givens rotation of points in the plane,
+		//
+		//  [x1; y1] := H[x1; y1].
+		y := []float64{test.Y1 * math.Sqrt(test.D2)}
```

Comment:
I think the `Sqrt` should not be here based at least on the rest of the test. We have `test.Rx1` that we want and `rx1` matches it. If I remove the sqrt, `x[0]` also matches it.

I will also look at this but I already concur that the documentation is quite bad.

Reply:
Yes, that's what I had previously, but that is at odds with the documentation. If I had some basis for removing the sqrt other than "it works better that way", I'd be pretty happy to make the change. They we can get down to fixing the single failing case.

Reply:
The BLAS reference paper clarifies this. The input is a scaling matrix D defined by d1, d2, and a vector [x1,y1], and H rotates [x1,y1] (to zero out y1).

Reply:
Any idea where the sqrts that are ubiquitous in the various drotmg documentation pages come from?

Reply:
The inputs to Drotmg define a vector `[a,b]^T` as
```
[a] = [ sqrt(d1)     0 ] * [ x1 ]
[b]   [ 0     sqrt(d2) ]   [ y1 ]
```
Drotmg rotates `[ x1, y1 ]^T` and updates the scaling. It could work directly with the square roots but only squares appear in the formulas, so it's more economical to define d1,d2 as squares of the actual coefficients. Taking the square root is necessary only when you want to form the vector `[a,b]^T` which is not the case in our tests.

Also, regarding the value of `gam`, in the Hanson-Hopkins reference you can read that the value of 4096 is very conservative and can be chosen much larger. This would reduce the occurence of rescaling which is good because the main saving in arithmetic operations comes from Drotm that uses the fact that H has normally two units (and rescaling spoils that). Except that I've just noticed that our Drotm does not! Feel free to create an issue, otherwise I'll do it tomorrow.

This `gam` value story also means that our `blas.Rescaling` test cases rely completely on the hardcoded value. If `gam` is as large as suggested, rescaling would maybe not take place and the test case would fail even though the result would be correct. We could modify `gam` which would support the reason of existence of Drotm(g) (performance) but then we would probably deviate from all other (public) blas implementations out there.

I haven't studied yet why the one test case fails, I may (or may not) have time for it tomorrow.